src/main/java/io/cdap/plugin/gcp/gcs/sink/GCSMultiBatchSink.java

```diff
@@ -92,12 +99,32 @@ public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
 
     PluginProperties formatProperties = formatPropertiesBuilder.build();
 
-    OutputFormatProvider outputFormatProvider =
-      pipelineConfigurer.usePlugin(ValidatingOutputFormat.PLUGIN_TYPE, format, FORMAT_PLUGIN_ID, formatProperties);
-    if (outputFormatProvider == null) {
-      collector.addFailure(
-        String.format("Could not find the '%s' output format plugin.", format), null)
-        .withPluginNotFound(FORMAT_PLUGIN_ID, format, ValidatingOutputFormat.PLUGIN_TYPE);
+    if (!this.config.containsMacro("format")) {
+      String format = config.getFormatName();
+      OutputFormatProvider outputFormatProvider =
+              pipelineConfigurer.usePlugin(ValidatingOutputFormat.PLUGIN_TYPE, format, FORMAT_PLUGIN_ID,
+                      formatProperties);
+      if (outputFormatProvider == null) {
+        collector.addFailure(
+                String.format("Could not find the '%s' output format plugin.", format), null)
+                .withPluginNotFound(FORMAT_PLUGIN_ID, format, ValidatingOutputFormat.PLUGIN_TYPE);
+      }
+    } else {
```

Comment:
Instead of else use return in the if statement. This will make the code more readable. Add a comment `deploying all format plugins if its macro so that required format plugin is available when macro is resolved in prepareRun`.

Reply:
the method doesn't have a return type, so I doubt 'if and return' can be applied. Please confirm

Reply:
What I mean was to change 
```
if (condition) {
// do something
} else {
// do something
}
```
To
```
if (condition) {
// do something
return;
}
// do something
```

Reply:
done